File: nodeJs/geraentradas2.js
fs=require('fs');

amostras=[];

//30 laranjas
for(x=0;x<30;x++){
	amostras.push({
		peso:0.8+Math.random()/10,			//entre 0.8 e 0.9
		volume:0.7+Math.random()/10,		//entre 0.7 e 0.8
		cor:0.6+Math.random()/10,			//entre 0.6 e 0.7
		posicao:0.5+Math.random()/10,		//entre 0.5 e 0.6
		textura:0.4+Math.random()/10,		//entre 0.4 e 0.5
		consistencia:0.3+Math.random()/10,	//entre 0.3 e 0.4
		temperatura:0.2+Math.random()/10,	//entre 0.2 e 0.3
		idade:0.1+Math.random()/10,			//entre 0.1 e 0.2
		especie:'l'							//laranja
	});
}

//30 frutas qualquer
for(x=0;x<30;x++){
	amostras.push({
		peso:Math.random()*2,				//entre 0 e 2
		volume:Math.random()*2,				//entre 0 e 2
		cor:Math.random()*2,				//entre 0 e 2
		posicao:Math.random()*2,			//entre 0 e 2
		textura:Math.random()*2,			//entre 0 e 2
		consistencia:Math.random()*2,		//entre 0 e 2
		temperatura:Math.random()*2,		//entre 0 e 2
		idade:Math.random()*2,				//entre 0 e 2
		especie:'q'							//uma fruta qualquer
	});
}

//salva como JSON
fs.writeFileSync('entradas2.txt',JSON.stringify(amostras),'utf-8');

//leitura e exibicao
lidas=JSON.parse(fs.readFileSync('entradas2.txt','utf-8'));
lidas.forEach((a,i)=>{

	if(a.especie=='l')a.especie='laranja';
	else a.especie='fruta qualquer';

	console.log(`${a.peso.toFixed(6)} ${a.volume.toFixed(6)} ${a.cor.toFixed(6)} ${a.posicao.toFixed(6)} ${a.textura.toFixed(6)} ${a.consistencia.toFixed(6)} ${a.temperatura.toFixed(6)} ${a.idade.toFixed(6)} ${a.especie}`);
});

File: python/geraentradas2.py
import random

with open("entradas2.txt","w") as f:
    #30 laranjas
    for x in range(30):
        especie='l'
        peso=0.8+random.random()/10.0 #entre 0.8 e 0.9
        volume=0.7+random.random()/10.0 #entre 0.7 e 0.8
        cor=0.6+random.random()/10.0 #entre 0.6 e 0.7
        posicao=0.5+random.random()/10.0 #entre 0.5 e 0.6
        textura=0.4+random.random()/10.0 #entre 0.4 e 0.5
        consistencia=0.3+random.random()/10.0 #entre 0.3 e 0.4
        temperatura=0.2+random.random()/10.0 #entre 0.2 e 0.3
        idade=0.1+random.random()/10.0 #entre 0.1 e 0.2
        f.write(f"{peso} {volume} {cor} {posicao} {textura} {consistencia} {temperatura} {idade} {especie}\n")
    #30 frutas quaisquer
    for x in range(30):
        especie='q'
        peso=random.random()*2.0 #entre 0 e 2
        volume=random.random()*2.0 #entre 0 e 2
        cor=random.random()*2.0 #entre 0 e 2
        posicao=random.random()*2.0 #entre 0 e 2
        textura=random.random()*2.0 #entre 0 e 2
        consistencia=random.random()*2.0 #entre 0 e 2
        temperatura=random.random()*2.0 #entre 0 e 2
        idade=random.random()*2.0 #entre 0 e 2
        f.write(f"{peso} {volume} {cor} {posicao} {textura} {consistencia} {temperatura} {idade} {especie}\n")

with open("entradas2.txt","r")as f:
    for line in f:
        parts=line.strip().split()
        inputs=[float(parts[i]) for i in range(8)]

        if parts[8]=='l':parts[8]='laranja'
        else: parts[8]='fruta qualquer'

        print(f"{inputs[0]:.6f} {inputs[1]:.6f} {inputs[2]:.6f} {inputs[3]:.6f} {inputs[4]:.6f} {inputs[5]:.6f} {inputs[6]:.6f} {inputs[7]:.6f} {parts[8]}")
